dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/WorkflowResource.java

```diff
@@ -622,7 +623,66 @@ public final Response deleteAction(@Context final HttpServletRequest request,
 
         return response;
     } // deleteAction
+    
+    /**
+     * Change the order of the steps in a scheme
+     * @param request                           HttpServletRequest
+     * @param workflowReorderActionStepForm     WorkflowReorderBean
+     * @return Response
+     */
+    @PUT
+    @Path("/reorder/step/{stepId}/order/{order}")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response reorderStep(@Context final HttpServletRequest request,
+                                        @PathParam("stepId")   final String stepId, 
+                                        @PathParam("order")    final int order) {
+
+        final InitDataObject initDataObject = this.webResource.init
+                (null, true, request, true, null);
+        Response response;
 
+        try {
+            WorkflowStep step = workflowAPI.findStep(stepId);
+            WorkflowScheme scheme = workflowAPI.findScheme(step.getSchemeId());
+            List<WorkflowStep> steps = workflowAPI.findSteps(scheme);
+            IntStream.range(0, steps.size())
+                .filter(i -> steps.get(i).getId().equals(step.getId()))
+                .boxed()
+                .findFirst()
+                .map(i -> steps.remove((int) i));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The return value of "map" must be used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2201)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Specify a type for: 'i' [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2211)